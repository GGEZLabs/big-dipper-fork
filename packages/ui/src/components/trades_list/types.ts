import { Trades } from '@/graphql/types/general_types';

export type TradesListState = {
  className?: string;
  hasNextPage?: boolean;
  isNextPageLoading?: boolean;
  loadNextPage?: (any) => void;
  loadMoreItems?: (any) => void;
  isItemLoaded?: (index: number) => boolean;
  itemCount?: number;
  trades: Trades[];
};
