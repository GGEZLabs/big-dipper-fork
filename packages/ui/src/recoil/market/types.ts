export type AtomState = {
  price: number | null;
  supply: TokenUnit;
  marketCap: number | null;
  inflation: number;
  communityPool: TokenUnit;
  apr: number;
  votingPrice: number | null;
  votingSupply: TokenUnit;
  votingMarketCap: number | null;
  votingInflation: number;
  votingCommunityPool: TokenUnit;
  votingApr: number;
};
