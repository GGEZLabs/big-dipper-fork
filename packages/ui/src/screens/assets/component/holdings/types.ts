import { Holding } from '@/graphql/types/general_types';

export type TransactionsListState = {
  className?: string;
  hasNextPage?: boolean;
  isNextPageLoading?: boolean;
  loadNextPage?: (any) => void;
  loadMoreItems?: (any) => void;
  isItemLoaded?: (index: number) => boolean;
  itemCount: number;
  transactions: Transactions[];
};

export type HoldingsListState = {
  className?: string;
  hasNextPage?: boolean;
  isNextPageLoading?: boolean;
  loadNextPage?: (any) => void;
  loadMoreItems?: (any) => void;
  isItemLoaded?: (index: number) => boolean;
  itemCount?: number;
  holdings: Holding[];
};
export type HoldingsState = {
  loading: boolean;
  exists: boolean;
  hasNextPage: boolean;
  isNextPageLoading: boolean;
  items: Holding[];
};
