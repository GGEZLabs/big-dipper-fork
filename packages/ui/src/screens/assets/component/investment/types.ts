export type TradeType = {
  id: number;
  asset_ticker: string;
  fund_name: string;
  timestamp: string;
  type: string;
  no_shares: number;
  currency: string;
  price: string;
  value: string;
  fee: string;
  net_value: string;
  net_price: string;
  issued_coins: number;
  coin_minting_price: number;
};

export type TradesState = {
  items: TradeType[];
};

// export type ItemType = Override<TradeType>
