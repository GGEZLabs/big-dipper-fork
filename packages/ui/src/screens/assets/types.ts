export type PortfolioPriceType = {
  time: string;
  value: number;
};

export type HeroState = {
  loading: boolean;
  exists: boolean;
  portfolioPriceHistory: PortfolioPriceType[];
};

export type AssetsIssuerGraphsType = {
  issuer: string;
  value: string;
  fill: string;
};

export type IssuerState = {
  loading: true;
  exists: true;
  items: AssetsIssuerGraphsType[];
};

export type AssetsSegmentGraphsType = {
  segment: string;
  value: string;
  fill: string;
};

export type SegmentState = {
  loading: true;
  exists: true;
  items: AssetsSegmentGraphsType[];
};
