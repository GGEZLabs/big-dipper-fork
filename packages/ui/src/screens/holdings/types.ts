import { Holding } from '@/graphql/types/general_types';

export type HoldingsState = {
  loading: boolean;
  exists: boolean;
  hasNextPage: boolean;
  isNextPageLoading: boolean;
  items: Holding[];
};
