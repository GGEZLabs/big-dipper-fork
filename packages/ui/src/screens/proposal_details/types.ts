export type OverviewType = {
  title: string;
  id: number | string;
  proposer: string;
  description: string;
  status: string;
  submitTime: string;
  depositEndTime: string;
  votingStartTime: string | null;
  votingEndTime: string | null;
  content: any;
};

export type ProposalState = {
  loading: boolean;
  exists: boolean;
  overview: OverviewType;
};
