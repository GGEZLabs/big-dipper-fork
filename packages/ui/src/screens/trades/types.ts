import { Trades } from '@/graphql/types/general_types';

export type HoldingsState = {
  loading: boolean;
  exists: boolean;
  hasNextPage: boolean;
  isNextPageLoading: boolean;
  items: Trades[];
};
