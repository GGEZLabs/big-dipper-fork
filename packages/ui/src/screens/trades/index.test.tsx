import { TradesListenerDocument, TradesDocument } from '@/graphql/types/general_types';
import Trades from '@/screens/trades';
import { mockClient } from '@/tests/mocks/mockApollo';
import MockTheme from '@/tests/mocks/MockTheme';
import wait from '@/tests/utils/wait';
import { ApolloProvider } from '@apollo/client';
import { MockedProvider } from '@apollo/client/testing';
import renderer from 'react-test-renderer';

// ==================================
// mocks
// ==================================
jest.mock('@/components/layout', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="Layout" {...props} />
));
jest.mock('@/components/trades_list', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="TradesList" {...props} />
));
jest.mock('@/components/box', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="Box" {...props} />
));
jest.mock('@/components/load_and_exist', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="LoadAndExist" {...props} />
));

const mockTradesListenerDocument = jest.fn().mockReturnValue({
  data: {
    trades: [
      {
        asset_ticker: 'CRBN',
        coin_minting_price: 1.0383e-8,
        currency: 'USD',
        fee: 21.56,
        fund_name: 'Low Carbon Target ETF',
        id: 9,
        issued_coins: 415371279977,
        net_price: 144.47866666666667,
        net_value: 4334.36,
        no_shares: 30,
        price: 143.76,
        type: 'buy',
        value: 4312.799999999999,
        timestamp: '2023-09-09T12:48:40.138522',
      },
    ],
  },
});

const mockTradesDocument = jest.fn().mockReturnValue({
  data: {
    trades: [
      {
        asset_ticker: 'CRBN',
        coin_minting_price: 1.0383e-8,
        currency: 'USD',
        fee: 21.56,
        fund_name: 'Low Carbon Target ETF',
        id: 9,
        issued_coins: 415371279977,
        net_price: 144.47866666666667,
        net_value: 4334.36,
        no_shares: 30,
        price: 143.76,
        type: 'buy',
        value: 4312.799999999999,
        timestamp: '2023-09-09T12:48:40.138522',
      },
      {
        asset_ticker: 'DSI',
        coin_minting_price: 1.0498e-8,
        currency: 'USD',
        fee: -75.49,
        fund_name: 'KLD 400 Social ETF',
        id: 8,
        issued_coins: 143817870070,
        net_price: 79.2645,
        net_value: -1585.29,
        no_shares: 20,
        price: 75.49,
        type: 'sell',
        value: -1509.8,
        timestamp: '2023-09-08T12:46:23.541618',
      },
      {
        asset_ticker: 'DSI',
        coin_minting_price: 1.0498e-8,
        currency: 'USD',
        fee: 3.77,
        fund_name: 'KLD 400 Social ETF',
        id: 7,
        issued_coins: 71908935035,
        net_price: 75.86699999999999,
        net_value: 758.67,
        no_shares: 10,
        price: 75.49,
        type: 'buy',
        value: 754.9,
        timestamp: '2023-09-07T12:42:44.103875',
      },
    ],
  },
});

let component: renderer.ReactTestRenderer | undefined;

// ==================================
// unit tests
// ==================================
describe('screen: Trades', () => {
  it('matches snapshot', async () => {
    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={mockClient}>
          <MockedProvider
            mocks={[
              {
                request: {
                  query: TradesListenerDocument,
                  variables: { limit: 1, offset: 0 },
                },
                result: mockTradesListenerDocument,
              },
              {
                request: { query: TradesDocument, variables: { limit: 3, offset: 0 } },
                result: mockTradesDocument,
              },
            ]}
          >
            <MockTheme>
              <Trades />
            </MockTheme>
          </MockedProvider>
        </ApolloProvider>
      );
    });
    await wait(renderer.act);
    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
