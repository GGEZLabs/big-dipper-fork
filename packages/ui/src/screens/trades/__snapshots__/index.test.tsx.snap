// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Trades matches snapshot 1`] = `
.emotion-0 {
  padding: 16px;
}

@media (min-width: 1280px) {
  .emotion-0 {
    padding: 16px 24px;
  }
}

.emotion-0 a {
  color: #1D86FF;
}

.emotion-1 {
  min-height: 500px;
  height: 50vh;
}

@media (min-width:1280px) {
  .emotion-1 {
    height: 100%;
    min-height: 65vh;
  }
}

<div
  className="emotion-0"
  id="Layout"
  navTitle="trades"
>
  <div
    exists={true}
    id="LoadAndExist"
    loading={false}
  >
    <div
      className="emotion-1"
      id="Box"
    >
      <div
        hasNextPage={false}
        id="TradesList"
        isItemLoaded={[Function]}
        isNextPageLoading={false}
        itemCount={1}
        loadMoreItems={[Function]}
        loadNextPage={[Function]}
        trades={
          [
            {
              "asset_ticker": "CRBN",
              "coin_minting_price": 1.0383e-8,
              "currency": "USD",
              "fee": 21.56,
              "fund_name": "Low Carbon Target ETF",
              "id": 9,
              "issued_coins": 415371279977,
              "net_price": 144.47866666666667,
              "net_value": 4334.36,
              "no_shares": 30,
              "price": 143.76,
              "timestamp": "2023-09-09T12:48:40.138522",
              "type": "buy",
              "value": 4312.799999999999,
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;
